### ISSUE TITLE
A false negative about the rule INT_BAD_COMPARISON_WITH_INT_VALUE

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#2147]
Hi, I found a false negative about  the rule [INT_BAD_COMPARISON_WITH_INT_VALUE](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#int-bad-comparison-of-int-value-with-long-constant-int-bad-comparison-with-int-value).

In the following code example, SpotBugs should have reported a warning at line 5, but it does not. Line 5 has a comparision between `int` and `long`, they are not matched. Hence, I think it is a false negaive of this rule. We can **analyze the data type of a method**, and the bug can be detected.

```java
long foo() {
  return 9999999999L;
}
public boolean badCheck(int x) {
  return x == foo();  // should report a warning at this line
}
```

SpotBugs version: 4.7.1

**Comment by ThrawnCA** [comment]
The rule isn't really designed to recognise situations where a constant is wrapped in a method call. What real world use case is there for this type of method?

**Comment by Zustin** [comment]
Got it, I thought it was caused by the type checking, thx.
